src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,11 @@
 package core.basesyntax;
 
 public class MainApp {
-
+    public static void main(String[] args) {
+        Machine[] machines = new Machine[] {new Truck(), new Bulldozer(), new Excavator()};
+        for (Machine machine : machines) {
+            machine.doWork();
+            machine.stopWork();
+        }
+    }
```

Comment:
The `MainApp` class implementation is concise and follows the Java OOP principles well. However, there is a missed opportunity to demonstrate informative messages within the `doWork` and `stopWork` methods. According to the checklist, it is important to ensure that these methods print out messages that indicate which type of machine is currently working or stopping. Make sure that the implementation of these methods in the respective machine classes includes such informative messages.